Installing on MacOsX Catalina

I can't seem to be able to install `python-poppler` on my system (MacOS X Catalina).
I have Xcode installed with the commandline utilities (I can compile with clang).
I run `brew install poppler` with no issue, the files are correctly found at `/usr/local/opt/poppler`.
My `CFLAGS` and `CPPFLAGS` environment variables both include `-I/usr/local/opt/poppler/include`.
I manage python2/3 via `pyenv`. `pyenv version` gives me `3.8.2` as expected.

When I run `pip install python-poppler` the installation fails.
Examining the log it seems that the compiler fails to find the header files:

```
    running build_ext
    -- The C compiler identification is AppleClang 11.0.3.11030032
    -- The CXX compiler identification is AppleClang 11.0.3.11030032
    -- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
    -- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - works
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
    -- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - works
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    -- Found PkgConfig: /usr/local/bin/pkg-config (found version "0.29.2")
    -- Checking for module 'poppler-cpp>=0.62.0'
    --   Found poppler-cpp, version 0.89.0
    -- Found PythonInterp: /Users/bordaigorl/.pyenv/versions/3.8.2/bin/python3.8 (found version "3.8.2")
    -- Found PythonLibs: /Users/bordaigorl/.pyenv/versions/3.8.2/lib/libpython3.8.a
    -- Performing Test HAS_CPP14_FLAG
    -- Performing Test HAS_CPP14_FLAG - Success
    -- pybind11 v2.5.0
    -- Performing Test HAS_FLTO
    -- Performing Test HAS_FLTO - Success
    -- LTO enabled
    -- Configuring done
    -- Generating done
    -- Build files have been written to: /private/var/folders/jx/r7g7g23n5rb_fdycrtdl5sdw0000gn/T/pip-install-j_u40vtr/python-poppler/build/temp.macosx-10.15-x86_64-3.8
    Scanning dependencies of target font
    Scanning dependencies of target page_renderer
    [  8%] Building CXX object CMakeFiles/page_renderer.dir/src/cpp/page_renderer.cpp.o
    [  8%] Building CXX object CMakeFiles/font.dir/src/cpp/font.cpp.o
    In file included from /private/var/folders/jx/r7g7g23n5rb_fdycrtdl5sdw0000gn/T/pip-install-j_u40vtr/python-poppler/src/cpp/page_renderer.cpp:19:
    /private/var/folders/jx/r7g7g23n5rb_fdycrtdl5sdw0000gn/T/pip-install-j_u40vtr/python-poppler/src/cpp/version.h:21:10: fatal error: 'poppler/cpp/poppler-version.h' file not found
    #include <poppler/cpp/poppler-version.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
```

I was very confused because `clang` sees the includes just fine if run manually.
I then thought that maybe it's cmake's fault and found this:
https://gitlab.kitware.com/cmake/cmake/-/issues/19120
So I symlinked

    sudo ln -s /usr/local/include/poppler /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/poppler

and now `pip install python-poppler` works.
Is there a way to fix this without dodgy symlinks?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,14 +14,12 @@ find_package(PkgConfig REQUIRED)
 pkg_check_modules(POPPLER REQUIRED IMPORTED_TARGET poppler-cpp>=0.62.0)
 
 add_library(global_ MODULE src/cpp/global.cpp)
-target_include_directories(global_ BEFORE PRIVATE ${POPPLER_INCLUDEDIR})
 target_link_libraries(global_ PRIVATE pybind11::module PkgConfig::POPPLER )
 set_target_properties(global_ PROPERTIES PREFIX "${PYTHON_MODULE_PREFIX}"
                                          SUFFIX "${PYTHON_MODULE_EXTENSION}")
 
 foreach (Target ${Targets})
     add_library(${Target} MODULE src/cpp/${Target}.cpp)
-    target_include_directories(${Target} BEFORE PRIVATE ${POPPLER_INCLUDEDIR})
     target_link_libraries(${Target} PRIVATE pybind11::module PkgConfig::POPPLER )
     set_target_properties(${Target} PROPERTIES PREFIX "${PYTHON_MODULE_PREFIX}"
                                                SUFFIX "${PYTHON_MODULE_EXTENSION}")

--- a/NEWS.txt
+++ b/NEWS.txt
@@ -3,6 +3,7 @@ Changelog
 0.2.1
 - Bugfix: Fixed byte order in image format string used for Pillow
 - Bugfix: Prevent segmentation fault when performing some operations on locked documents (fixes #4)
+- Bugfix: Fix include paths in C++ (fixes #2)
 
 0.2.0
 - Add the font infos to the TextBox object (Poppler 0.89.0)

--- a/src/cpp/destination.cpp
+++ b/src/cpp/destination.cpp
@@ -19,7 +19,7 @@
 #include "version.h"
 #include <pybind11/pybind11.h>
 #if HAS_VERSION(0, 74)
-#include <poppler/cpp/poppler-destination.h>
+#include <poppler-destination.h>
 #endif
 
 namespace py = pybind11;

--- a/src/cpp/document.cpp
+++ b/src/cpp/document.cpp
@@ -20,13 +20,13 @@
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
 #if HAS_VERSION(0, 74)
-#include <poppler/cpp/poppler-destination.h>
+#include <poppler-destination.h>
 #endif
-#include <poppler/cpp/poppler-document.h>
-#include <poppler/cpp/poppler-embedded-file.h>
-#include <poppler/cpp/poppler-font.h>
-#include <poppler/cpp/poppler-page.h>
-#include <poppler/cpp/poppler-toc.h>
+#include <poppler-document.h>
+#include <poppler-embedded-file.h>
+#include <poppler-font.h>
+#include <poppler-page.h>
+#include <poppler-toc.h>
 
 namespace py = pybind11;
 

--- a/src/cpp/embedded_file.cpp
+++ b/src/cpp/embedded_file.cpp
@@ -18,7 +18,7 @@
 
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
-#include <poppler/cpp/poppler-embedded-file.h>
+#include <poppler-embedded-file.h>
 
 namespace py = pybind11;
 

--- a/src/cpp/font.cpp
+++ b/src/cpp/font.cpp
@@ -18,7 +18,7 @@
 
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
-#include <poppler/cpp/poppler-font.h>
+#include <poppler-font.h>
 
 namespace py = pybind11;
 

--- a/src/cpp/global.cpp
+++ b/src/cpp/global.cpp
@@ -18,7 +18,7 @@
 
 #include "version.h"
 #include <pybind11/pybind11.h>
-#include <poppler/cpp/poppler-global.h>
+#include <poppler-global.h>
 #include <algorithm>
 
 namespace py = pybind11;

--- a/src/cpp/image.cpp
+++ b/src/cpp/image.cpp
@@ -19,8 +19,8 @@
 #include "version.h"
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
-#include <poppler/cpp/poppler-image.h>
-#include <poppler/cpp/poppler-rectangle.h>
+#include <poppler-image.h>
+#include <poppler-rectangle.h>
 
 namespace py = pybind11;
 

--- a/src/cpp/page.cpp
+++ b/src/cpp/page.cpp
@@ -19,10 +19,10 @@
 #include "version.h"
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
-#include <poppler/cpp/poppler-global.h>
-#include <poppler/cpp/poppler-page.h>
-#include <poppler/cpp/poppler-page-transition.h>
-#include <poppler/cpp/poppler-rectangle.h>
+#include <poppler-global.h>
+#include <poppler-page.h>
+#include <poppler-page-transition.h>
+#include <poppler-rectangle.h>
 
 namespace py = pybind11;
 

--- a/src/cpp/page_renderer.cpp
+++ b/src/cpp/page_renderer.cpp
@@ -18,10 +18,10 @@
 
 #include "version.h"
 #include <pybind11/pybind11.h>
-#include <poppler/cpp/poppler-page-renderer.h>
-#include <poppler/cpp/poppler-global.h>
-#include <poppler/cpp/poppler-image.h>
-#include <poppler/cpp/poppler-page.h>
+#include <poppler-page-renderer.h>
+#include <poppler-global.h>
+#include <poppler-image.h>
+#include <poppler-page.h>
 
 namespace py = pybind11;
 

--- a/src/cpp/page_transition.cpp
+++ b/src/cpp/page_transition.cpp
@@ -17,7 +17,7 @@
  */
 
 #include <pybind11/pybind11.h>
-#include <poppler/cpp/poppler-page-transition.h>
+#include <poppler-page-transition.h>
 
 namespace py = pybind11;
 

--- a/src/cpp/rectangle.cpp
+++ b/src/cpp/rectangle.cpp
@@ -17,7 +17,7 @@
  */
 
 #include <pybind11/pybind11.h>
-#include <poppler/cpp/poppler-rectangle.h>
+#include <poppler-rectangle.h>
 
 namespace py = pybind11;
 

--- a/src/cpp/toc.cpp
+++ b/src/cpp/toc.cpp
@@ -18,8 +18,8 @@
 
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
-#include <poppler/cpp/poppler-global.h>
-#include <poppler/cpp/poppler-toc.h>
+#include <poppler-global.h>
+#include <poppler-toc.h>
 
 namespace py = pybind11;
 

--- a/src/cpp/version.cpp
+++ b/src/cpp/version.cpp
@@ -17,7 +17,7 @@
  */
 
 #include <pybind11/pybind11.h>
-#include <poppler/cpp/poppler-version.h>
+#include <poppler-version.h>
 
 namespace poppler::version
 {

--- a/src/cpp/version.h
+++ b/src/cpp/version.h
@@ -18,7 +18,7 @@
 
 #pragma once
 
-#include <poppler/cpp/poppler-version.h>
+#include <poppler-version.h>
 
 
 #define HAS_VERSION(major,minor) (POPPLER_VERSION_MAJOR > major) || (POPPLER_VERSION_MAJOR == major && POPPLER_VERSION_MINOR >= minor)

